Source/AwsCommonRuntimeKit/io/TLSContextOptions.swift

```diff
@@ -9,10 +9,46 @@ public class TLSContextOptions: CStruct {
         TLSContextOptions()
     }
 
-    public static func makeMtlsPkcs12FromPath(
-        path: String,
+    /// Initializes TlsContextOptions for mutual TLS (mTLS), with client certificate and private key in the PKCS#12 format.
+    ///
+    /// NOTE: This only works on Apple devices.
+    ///
+    /// - Parameters:
+    ///     - pkcs12Path: Path to PKCS #12 file. The file is loaded from disk and stored internally. It must remain in
+    ///     memory for the lifetime of the returned object.
+    ///     - password: Password to PKCS #12 file. It must remain in memory for the lifetime of the returned object.
+    public static func makeMtls(
+        pkcs12Path: String,
         password: String) throws -> TLSContextOptions {
-        try TLSContextOptions(mtlsPkcs12FromPath: path, password: password)
+        try TLSContextOptions(mtlsPkcs12FromPath: pkcs12Path, password: password)
+    }
+
+    /// Initializes TlsContextOptions for mutual TLS (mTLS), with client certificate and private key. These are in memory
+    /// buffers. These buffers must be in the PEM format.
+    ///
+    /// NOTE: This is unsupported on iOS.
```

Comment:
Thinking about it, it's better to set restrictions early on instead of adding them later. I added the available here for the path and data API. 
```
    @available(iOS, unavailable)
    @available(tvOS, unavailable)
    @available(watchOS, unavailable)
    public static func makeMTLS(
        certificateData: String,
        privateKeyData: String) throws -> TLSContextOptions {
        try TLSContextOptions(certificateData: certificateData, privateKeyData: privateKeyData)
    }
```

For the pkcs12 one, it seems that `@available` only works for Apple platforms, we dont have a way to disable the API for linux/windows. I will leave it there for now.